infra/src/main/kotlin/fi/fta/geoviite/api/frameconverter/v1/FrameConverterServiceV1.kt

```diff
@@ -0,0 +1,343 @@
+                null
+            }
+        }
+
+        val locationTrackNameOrNull = request.locationTrackName?.let { unvalidatedLocationTrackName ->
+            try {
+                AlignmentName(unvalidatedLocationTrackName)
+            } catch (e: InputValidationException) {
+                errors.add("invalid-location-track-name")
+                null
+            }
+        }
+
+        val nonNullErrors = errors.filterNotNull()
+        return if (nonNullErrors.isEmpty()) {
+            // Already checked earlier but type-inference is not smart enough =(
+            requireNotNull(request.x)
+            requireNotNull(request.y)
+            requireNotNull(request.searchRadius)
+
+            val validRequest = ValidCoordinateToTrackMeterRequestV1(
+                request.identifier,
+                request.x,
+                request.y,
+                request.searchRadius,
+                trackNumberName = trackNumberNameOrNull,
+                locationTrackName = locationTrackNameOrNull,
+                locationTrackType = mappedLocationTrackTypeOrNull,
+                request.responseSettings,
+            )
+
+            validRequest to emptyList()
+        } else {
+            val errorResponse = createErrorResponse(
+                identifier = request.identifier,
+                errorLocalizationKeys = nonNullErrors,
+            )
+
+            null to errorResponse
+        }
+    }
+
+    fun createErrorResponse(
+        identifier: String?,
+        errorLocalizationKey: String
+    ) = createErrorResponse(identifier, listOf(errorLocalizationKey))
+
+    fun createErrorResponse(
+        identifier: String?,
+        errorLocalizationKeys: List<String>
+    ): List<GeoJsonFeatureErrorResponseV1> {
+        return listOf(
+            GeoJsonFeatureErrorResponseV1(
+                identifier = identifier,
+                errorMessages = errorLocalizationKeys.joinToString(" ") { localizationKey ->
+                    translation.t("integration-api.error.$localizationKey")
+                }
+            )
+        )
+    }
+
+    private fun translateLocationTrackType(locationTrack: LocationTrack): String {
+        return translation.t("enum.location-track-type.${locationTrack.type}")
+    }
+
+    private fun filterByLocationTrackType(
+        locationTrackType: LocationTrackType?,
+        locationTrack: LocationTrack,
+    ): Boolean {
+        return if (locationTrackType == null) {
+            true
+        } else {
+            locationTrackType == locationTrack.type
+        }
+    }
+
+    private fun filterByTrackNumber(
+        layoutContext: LayoutContext,
+        requestTrackNumber: TrackNumber?,
+        locationTrack: LocationTrack,
+    ): Boolean {
+        return if (requestTrackNumber == null) {
+            true
+        } else {
+            val trackNumber = trackNumberService.get(layoutContext, locationTrack.trackNumberId)
+            requestTrackNumber == trackNumber?.number
```

Comment:
Aiheuttaakohan perffiongelmia hakea näitä jokaisella chekillä erikseen? Toistaiseksi siis pyritty (mahdollisimman) yksinkertaiseen koodiin.